### ISSUE TITLE
got ERR_OSSL_EVP_UNSUPPORTED when using node 17 and react 17

### ISSUE THREAD

**Issue by rafipiccolo** [expo/expo-cli#3943]
### Summary

when i run expo start it returns a qrcode and then shows this error : ERR_OSSL_EVP_UNSUPPORTED
i can boot phone, but it show a second error and nothing works

### Environment

  Expo CLI 4.12.10 environment info:
    System:
      OS: Linux 5.4 Ubuntu 20.04.3 LTS (Focal Fossa)
      Shell: 5.0.17 - /bin/bash
    Binaries:
      Node: 17.0.1 - /usr/local/bin/node
      Yarn: 1.22.15 - /usr/local/bin/yarn
      npm: 8.1.1 - /usr/local/bin/npm
    npmPackages:
      expo: ^43.0.0 => 43.0.0 
      react: 17.0.1 => 17.0.1 
      react-dom: 17.0.1 => 17.0.1 
      react-native: 0.64.2 => 0.64.2 
    npmGlobalPackages:
      expo-cli: 4.12.10
    Expo Workflow: managed

### Please specify your device/emulator/simulator platform, model and version

android 11 - one plus 6T

### Error output

```
# expo start
...
Logs for your project will appear below.
Failed to construct transformer:  Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:67:19)
    at Object.createHash (node:crypto:130:10)
    at stableHash (/root/docker/gextrareact/node_modules/metro-cache/src/stableHash.js:19:8)
    at Object.getCacheKey (/root/docker/gextrareact/node_modules/metro-transform-worker/src/index.js:647:7)
    at getTransformCacheKey (/root/docker/gextrareact/node_modules/metro/src/DeltaBundler/getTransformCacheKey.js:32:19)
    at new Transformer (/root/docker/gextrareact/node_modules/metro/src/DeltaBundler/Transformer.js:180:28)
    at /root/docker/gextrareact/node_modules/metro/src/Bundler.js:58:29 {
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}

# and when opening phone :
Android Bundling failed 88ms
Cannot read properties of undefined (reading 'transformFile')
```

### Reproducible demo or steps to reproduce from a blank project

expo init my-app
cd my-app
expo start

**Comment by brentvatne** [comment]
we recommend using one of the node LTS releases for the best experience: https://docs.expo.dev/get-started/installation/#requirements

if anyone wants to dig into the root cause here it would be helpful!

**Comment by kbrandwijk** [comment]
The Metro bundler is not compatible yet with Node 17.

While there is a workaround, which is running `webpack` with the `openssl-legacy-provider` option, this option _only_ works on Node 17, would be temporary, and would need to be deeply embedded in the CLI.

To do this manually, run `export NODE_OPTIONS=--openssl-legacy-provider`. After this, `expo start` will work fine on Node 17.

**Comment by kbrandwijk** [comment]
The Metro bundler version is linked to the RN release. Depending on which RN version this fix lands in, it might or might not be part of the next SDK.

**Comment by brentvatne** [comment]
it looks like this should indeed be fixed in sdk 45 / rn 0.68 because the fix included in metro@0.67 (which is used by rn 0.68)

**Comment by Jamal-ReachFirst** [comment]
Reopen
